cmis/unit/backend_adapter.py

```diff
@@ -1,61 +1,34 @@
 # -*- coding: utf-8 -*-
-##############################################################################
-#
-#    OpenERP, Open Source Management Solution
-#    This module copyright (C) 2015 - Present Savoir-faire Linux
-#    (<http://www.savoirfairelinux.com>).
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as
-#    published by the Free Software Foundation, either version 3 of the
-#    License, or (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program. If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2014-2015 Savoir-faire Linux (<http://www.savoirfairelinux.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from openerp.osv import orm
-from openerp.tools.translate import _
-from openerp.addons.connector.unit.backend_adapter import CRUDAdapter
-import cmislib.exceptions
 from cmislib.model import CmisClient
+import cmislib.exceptions
 import urllib2
 
+from openerp.tools.translate import _
+from openerp.addons.connector.unit.backend_adapter import CRUDAdapter
+from openerp.addons.cmis.exceptions import CMISConnectionError
+
 
 class CmisAdapter(CRUDAdapter):
 
-    def _auth(self, ids):
+    def _auth(self, cmis_backend):
         """Test connection with CMIS"""
-        if type(ids) is not list:
-            ids = [ids]
-        # Get the url, user and password for CMIS
-        # ids = self.search(cr, uid, [])
-        if not ids:
-            raise orm.except_orm(
-                _('Internal Error'),
-                _('Something very wrong happened. _auth() '
-                    'called without any ids.')
-            )
-        res = self.backend_record.read(['location', 'username', 'password'])[0]
-        url = res['location']
-        user_name = res['username']
-        user_password = res['password']
-        client = CmisClient(url, user_name, user_password)
+        cmis_backend.ensure_one()
+        client = CmisClient(
+            cmis_backend.location,
+            cmis_backend.username,
+            cmis_backend.password)
 
         try:
             return client.defaultRepository
```

Comment:
shouldn't you return only the client object. Returning a repository seems out of bound for a method named _auth.